### ISSUE TITLE
Add spec.ingressClassName to kiali-server/templates/ingress.yaml

### ISSUE THREAD

**Issue by Christoph-Raab** [kiali/kiali#4455]
The [ingress.yaml](https://github.com/kiali/helm-charts/blob/master/kiali-server/templates/ingress.yaml) of the kiali-server helm chart provides the possibility to pass the `kubernetes.io/ingress.class` annotation via `.Values.deployment.override_ingress_yaml.metadata.annotations`. Many ingress controller rely on that annotation.

Kubernetes now [deprecated](https://kubernetes.io/docs/concepts/services-networking/ingress/#deprecated-annotation) this annotation and recommands to use `spec.ingressClassName`.

With the current setup one can only provide a full spec of the ingress or use the default. This basically makes the default ingress unusable for anyone that relies on that field.

A possible solution would be to add the field to the `override_ingress_yaml`:

```
[...]
spec:
  {{- if hasKey .Values.deployment.override_ingress_yaml "spec" }}
  {{- toYaml .Values.deployment.override_ingress_yaml.spec | nindent 2 }}
  {{- else }}
  {{- if hasKey .Values.deployment.override_ingress_yaml "ingressClassName" }}
  ingressClassName: {{ .Values.deployment.override_ingress_yaml.ingressClassName }}
  {{- end }}
  rules:
  - http:
      paths:
[...]
```

By that anyone can set the `ingressClassName` and still use the default ingress resource of the helm chart:

```
[...]
override_ingress_yaml:
  metadata:
    labels:
      some-label: some-value
    annotations:
      ingress.kubernetes.io/ssl-redirect: "true"
  ingressClassName: traefik
[...]
```

I am happy to create a PR into the [helm charts](https://github.com/kiali/helm-charts) repository if you're if you are satisfied with the proposal.

**Comment by jmazzitelli** [comment]
for those interested in this issue, see: https://github.com/kiali/kiali/issues/4342#issuecomment-956300708

**Comment by jmazzitelli** [comment]
this is now fixed - it will be released with 1.43 next week. Please see: https://github.com/kiali/kiali/issues/4342#issuecomment-956300708
